bba/bba_internal_test.go

```diff
@@ -0,0 +1,435 @@
+package bba
+
+import (
+	"encoding/json"
+	"strconv"
+	"testing"
+
+	"github.com/DE-labtory/cleisthenes/pb"
+
+	"github.com/DE-labtory/cleisthenes"
+	"github.com/DE-labtory/cleisthenes/test/mock"
+)
+
+type bbaTester struct {
+	assertMap map[int]func(t *testing.T, bbaInstance *BBA)
+}
+
+func newBbaTester() *bbaTester {
+	return &bbaTester{
+		assertMap: make(map[int]func(t *testing.T, bbaInstance *BBA)),
+	}
+}
+
+func (h *bbaTester) setupBvalRequestList(bvalList []*bvalRequest) ([]request, error) {
+	result := make([]request, 0)
+	for i, bval := range bvalList {
+		d, err := json.Marshal(bval)
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, request{
+			sender: cleisthenes.Member{Address: cleisthenes.Address{Ip: "localhost" + strconv.Itoa(i), Port: 8080}},
+			data: &pb.Message_Bba{
+				Bba: &pb.BBA{
+					Type:    pb.BBA_BVAL,
+					Payload: d,
+				},
+			},
+		})
+	}
+	return result, nil
+}
+
+func (h *bbaTester) setupAuxRequestList(bvalList []*auxRequest) ([]request, error) {
+	result := make([]request, 0)
+	for i, bval := range bvalList {
+		d, err := json.Marshal(bval)
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, request{
+			sender: cleisthenes.Member{Address: cleisthenes.Address{Ip: "localhost" + strconv.Itoa(i), Port: 8080}},
+			data: &pb.Message_Bba{
+				Bba: &pb.BBA{
+					Type:    pb.BBA_AUX,
+					Payload: d,
+				},
+			},
+		})
+	}
+	return result, nil
+}
+
+func (h *bbaTester) setupRequestRepository() cleisthenes.RequestRepository {
```

Comment:
`requestRepository` is simple to implements. I think implement `Repository`s is better than mocking.